portfolio/src/main/java/com/google/sps/servlets/HelloWorldServlet.java

```diff
@@ -1,18 +1,50 @@
 package com.google.sps.servlets;
 
 import java.io.IOException;
+import java.util.ArrayList;
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
+import com.google.gson.*;
 
 /** Handles requests sent to the /hello URL. Try running a server and navigating to /hello! */
 @WebServlet("/hello")
 public class HelloWorldServlet extends HttpServlet {
-
+    
   @Override
-  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("text/html;");
-    response.getWriter().println("Here's a fetch response!");
+  public void doGet(final HttpServletRequest request, final HttpServletResponse response) throws IOException {
+    ArrayList<String> randFacts = new ArrayList<String>();
+    randFacts.add("I was born somewhere between Laos and Thailand while my parents were en route to a refugee camp.");
+    randFacts.add("I recently moved to Fresno, CA from Visalia, CA and I'm living with 2 dogs, Max and Mister.");
+    randFacts.add("I love dogs (and i'm also allergic to them).");
+    randFacts.add("My favorite food is shrimp. I will eat any dish if it's included.");
+    String json = convertToJson(randFacts);
```

Comment:
Don't use convertToJson. This is brittle. For example if you added another fact to `randFacts` it wouldn't be included in the response, and if you deleted one your server would crash on `json += "\"" + randFacts.get(3) + "\"";` because randFacts doesn't have a 4th object.

I think part of the exercise was meant to show how difficult it is to create your own json strings dynamically. Rely on the `Gson` library instead - they've done the hard work for you 😄

Reply:
Make sure you're not relying on any of the Json object keys in the web page though.